tasks/main.yml

```diff
@@ -24,9 +24,21 @@
     ( ansible_distribution_major_version > "7" and
       ( ansible_distribution == "CentOS" or ansible_distribution == "RedHat" ))
 
-- name: Set permanent SELinux state
-  selinux: policy={{ selinux_policy }} state={{ selinux_state }}
-  when: selinux_state is defined
+- name: Set permanent SELinux state if enabled
+  selinux:
+    state: "{{ selinux_state | default(ansible_selinux.config_mode) }}"
+    policy: "{{ selinux_policy | default(ansible_selinux.type) }}"
+  when: ansible_selinux.status == "enabled"
```

Comment:
If SELinux is enabled, `selinux` will be always run even when it does not change anything because it uses current values.

What about:

`when: ansible_selinux.status == "enabled" and ( selinux_state is defined or selinux_state is defined )`

Reply:
Fixed

Reply:
> If SELinux is enabled, selinux will be always run even when it does not change anything because it uses current values.

Not entirely true. It uses current _configured_ values (`config_mode` which can be different from the running configuration (`mode`).

Reply:
@bachradsusi I actually think you were right and one should not call the module when not `( selinux_state is defined or selinux_policy is defined )`. Not because it would not change anything (that would be an unimportant optimization), but because it could change something (reset the running state to the saved state if they differ) and such change would be unexpected if one has not defined `selinux_state` or `selinux_policy`.

I filed #24 to return to your proposal (with @wrabcak 's correction).